### PR TITLE
Moving most of generate_dictionary into the library.

### PR DESCRIPTION
**Pull request by Yoric**
We will use this in followups to script auto-generation of dictionaries during benchmarking.

### CODE REVIEW

File: src/cli/generate_dictionary.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        return Ok(());
+    }
+
+    progress!(options.quiet, "Parsing.");
+
+    handle_path_or_text(options, shared_builder, shared_number_of_files, source_path)
+}
+
+fn handle_path_or_text<'a>(
+    options: &mut Options<'a>,
```
Comment (RReverser) [inline src/cli/generate_dictionary.rs]: Why does this need to be mutable?

File: src/cli/generate_dictionary.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    dictionary_builder: &mut DictionaryBuilder,
+    shared_number_of_files: &mut usize,
```
Comment (RReverser) [inline src/cli/generate_dictionary.rs]: Given that this is always incremented on success, perhaps worth removing a param here and keep track of count in the caller instead?

File: src/cli/generate_dictionary.rs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        let sub_dir = sub_dir.join(file_name);
+        for entry in std::fs::read_dir(source_path)?
+            .map(|dir| dir.unwrap())
+        {
+            handle_path(
+                options,
+                shared_builder,
+                shared_number_of_files,
+                &entry.path().as_path(),
```
Comment (RReverser) [inline src/cli/generate_dictionary.rs]: I don't think you need `.as_path()` here - `&PathBuf` already coerces to `&Path`.

**Comment by Yoric** [comment]
Killing this off in favor of #272 .

**Pull request closed without merging**